v3.9.0 regression - panic in getNamespacesForRoleBinding

Our builds have just started running with kustomize v3.9.0 and are panicking. This can only be a regression.

https://github.com/argoproj/argo/pull/4618/checks?check_run_id=1533236396

```
panic: interface conversion: interface {} is string, not map[string]interface {}

goroutine 1 [running]:
sigs.k8s.io/kustomize/api/resmap.getNamespacesForRoleBinding(0xc0009774a0, 0x19)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/resmap/reswrangler.go:417 +0x33f
sigs.k8s.io/kustomize/api/resmap.(*resWrangler).SubsetThatCouldBeReferencedByResource(0xc00044bd60, 0xc0009774a0, 0xb, 0xc0028a9501)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/resmap/reswrangler.go:384 +0xca
sigs.k8s.io/kustomize/api/internal/accumulator.(*nameReferenceTransformer).Transform(0xc00243edc0, 0x1990300, 0xc00044bd60, 0xc00044bd60, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/namereferencetransformer.go:85 +0x515
sigs.k8s.io/kustomize/api/internal/accumulator.(*ResAccumulator).Transform(...)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/resaccumulator.go:138
sigs.k8s.io/kustomize/api/internal/accumulator.(*ResAccumulator).FixBackReferences(0xc00044bd40, 0xc00044bd40, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/internal/accumulator/resaccumulator.go:164 +0xba
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).makeCustomizedResMap(0xc000206a80, 0x0, 0x0, 0x1986bc0, 0x25be0e8)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/internal/target/kusttarget.go:125 +0x7b
sigs.k8s.io/kustomize/api/internal/target.(*KustTarget).MakeCustomizedResMap(...)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/internal/target/kusttarget.go:106
sigs.k8s.io/kustomize/api/krusty.(*Kustomizer).Run(0xc000e77d20, 0x7ffd0447e18f, 0x19, 0x0, 0x0, 0x0, 0x0)
	/go/pkg/mod/sigs.k8s.io/kustomize/api@v0.7.0/krusty/kustomizer.go:77 +0x294
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.(*Options).RunBuild(0xc00015bb90, 0x192dde0, 0xc00000e018, 0x2, 0x0)
	/workspace/myClone/kustomize/internal/commands/build/build.go:148 +0xb5
sigs.k8s.io/kustomize/kustomize/v3/internal/commands/build.NewCmdBuild.func1(0xc000218580, 0xc00044aba0, 0x1, 0x2, 0x0, 0x0)
	/workspace/myClone/kustomize/internal/commands/build/build.go:70 +0xab
github.com/spf13/cobra.(*Command).execute(0xc000218580, 0xc00044ab80, 0x2, 0x2, 0xc000218580, 0xc00044ab80)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000bd8c0, 0x0, 0x14d5840, 0xc00007e058)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
	/workspace/myClone/kustomize/main.go:19 +0x27
make: *** [manifests/install.yaml] Error 2
```
<!-- A clear and concise description of what the bug is. -->

**Files that can reproduce the issue**

<!--
We cannot figure out or fix the issue if we don't know how to reproduce. Please
provide a minimum set of files that can reproduce the issue. You can paste the
file contents here or provide a link to a tarball or git repo.

Example:

kustomization.yaml

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
...
```

resources.yaml

```
apiVersion: v1
kind: Deployment
...
```

...
-->

**Expected output**

<!-- What's the expected output? -->

**Actual output**

<!-- What's the actual output? -->

**Kustomize version**

<!-- Please use the latest version when it's possible. -->

**Platform**

<!-- Linux/macOS/Windows? -->

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/api/internal/wrappy/wnode.go
+++ b/api/internal/wrappy/wnode.go
@@ -91,8 +91,8 @@ func (wn *WNode) GetFieldValue(path string) (interface{}, error) {
 	// Return value as slice for SequenceNode kind
 	if yn.Kind == yaml.SequenceNode {
 		var result []interface{}
-		for _, node := range yn.Content {
-			result = append(result, node.Value)
+		if err := yn.Decode(&result); err != nil {
+			return nil, err
 		}
 		return result, nil
 	}

--- a/api/internal/wrappy/wnode_test.go
+++ b/api/internal/wrappy/wnode_test.go
@@ -377,6 +377,39 @@ func TestGetFieldValueReturnsSlice(t *testing.T) {
 	}
 }
 
+func TestGetFieldValueReturnsSliceOfMappings(t *testing.T) {
+	bytes, err := yaml.Marshal(makeBigMap())
+	if err != nil {
+		t.Fatalf("unexpected yaml.Marshal err: %v", err)
+	}
+	rNode, err := kyaml.Parse(string(bytes))
+	if err != nil {
+		t.Fatalf("unexpected yaml.Marshal err: %v", err)
+	}
+	wn := FromRNode(rNode)
+	expected := []interface{}{
+		map[string]interface{}{
+			"field1": "idx0foo",
+			"field2": "idx0bar",
+		},
+		map[string]interface{}{
+			"field1": "idx1foo",
+			"field2": "idx1bar",
+		},
+		map[string]interface{}{
+			"field1": "idx2foo",
+			"field2": "idx2bar",
+		},
+	}
+	actual, err := wn.GetFieldValue("those")
+	if err != nil {
+		t.Fatalf("error getting slice: %v", err)
+	}
+	if diff := cmp.Diff(expected, actual); diff != "" {
+		t.Fatalf("actual slice does not deep equal expected slice:\n%v", diff)
+	}
+}
+
 func TestGetFieldValueReturnsString(t *testing.T) {
 	wn := NewWNode()
 	if err := wn.UnmarshalJSON([]byte(deploymentBiggerJson)); err != nil {

--- a/kustomize/go.mod
+++ b/kustomize/go.mod
@@ -19,3 +19,5 @@ exclude (
 	sigs.k8s.io/kustomize/api v0.2.0
 	sigs.k8s.io/kustomize/cmd/config v0.2.0
 )
+
+replace sigs.k8s.io/kustomize/api v0.7.0 => ../api

